azurelinuxagent/common/utils/restutil.py

```diff
@@ -312,11 +312,17 @@ def _http_request(method, host, rel_uri, port=None, data=None, secure=False, # p
                                          conn_port,
                                          timeout=10)
 
+
+    if method == "PUT" and "vmAgentLog" in url:
```

Comment:
I dont particularly like this condition here. What's the use case? Since we're just logging it to the verbose log, does it really matter if we redact it or not?

Reply:
Synced offline, using a flag approach.